### PR TITLE
Add more interpolator examples

### PR DESCRIPTION
**Pull request by tommy-gilligan**
This adds Rust translations of some of the official interpolator examples:

- linear_interpolation
- simple_blend3

I think the interpolator can be a bit unusual and so having lots of examples for it is a good thing.

The simple_blend3 example motivated the work at https://github.com/rp-rs/rp-hal/pull/543 (Sorry I didn't reply before merge.  Did not want to ping unnecessarily by replying in thread.  Thank you @jannic ! )

Dependent on unreleased v0.8 of rp2040-hal https://github.com/rp-rs/rp-hal-boards/pull/10

### CODE REVIEW

**Comment by 9names** [comment]
We've typically tried to keep examples that rely on external hardware in rp-pico instead of being in rp-hal.
That said: I think a (single, non-smart) LED is close enough to a GPIO pin that I would have no objections to this being in rp-hal - as you say, it's a complex peripheral, and we *do* want good core usage examples for peripherals in the hal itself.

File: boards/rp-pico/examples/pico_interpolator.rs [hunk 1]
```diff
@@ -357,6 +390,89 @@ fn clamp(interp: &mut Interp1) -> bool {
...
+        (-1000, 0, -1000, 500),
+        (-625, 25, -1000, 500),
+        (-250, 50, -1000, 500),
+        (125, 75, -1000, 500),
+    ];
+    for (expected_peek, expected_percent, expected_pair_0, expected_pair_1) in expected {
+        // result2 = samples + (lane0 raw result)
+        // i.e. ptr to the first of two samples to blend between
+        let sample_pair: [i16; 2] = unsafe {
+            let sp: &[i16] = &(interp.peek() as *const [i16; 5]).read()[0..2];
```
Comment (jannic) [inline boards/rp-pico/examples/pico_interpolator.rs:450]: This example is really hard to follow. I wondered why this `unsafe` block is fine (and there should really be a comment explaining that). With some help from the datasheet I was able to infer how it works. I'd prefer if the examples were self-explaining and easy to understand.

Also I'm not sure if it's a great example, arithmetic on raw pointers isn't very usual in Rust programs. It might provide a performance benefit to let the interpolator do the whole calculation in one step, but that also should be justified by a comment, and ideally be verified by some benchmark or assembly analysis.

And, if I understand correctly what happens here (I'm not 100% sure!), this code has a bug and will trigger undefined behavior: `interp.peek() as *const [i16; 5]` is a pointer to an array `[i16; 5]`, but it points to some location within the original `samples` array. `read()` will read 5 `i16` values starting from that location, that is, it will read past the end of `samples`, which is not allowed. Even if then, only the first two of those values are actually used.
